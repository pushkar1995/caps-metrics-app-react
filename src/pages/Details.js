import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getCountry } from '../redux/countriesSlice/countriesSlice';
import Navbar from '../components/Navbar/Navbar';
import './Details.scss';

const Details = () => {
  const { countrySearch } = useSelector((store) => store.country);
  const dispatch = useDispatch();
  const { code } = useParams();

  useEffect(() => {
    if (code) {
      dispatch(getCountry(code.toLowerCase()));
    }
  }, [dispatch, code]);
  return (
    <div>
      <Navbar title="Details" back />
      {countrySearch.length > 0 ? (
        <div className="detail__container">
          <div className="detail__header">
            <img
              src={countrySearch[0].flags.png}
              alt={countrySearch[0].flags.alt}
            />
            <div>
              <h2>{countrySearch[0].name.official}</h2>
              <p>{countrySearch[0].region}</p>
            </div>
          </div>
          <div className="detail__description">
            <p>
              <span>Common name</span>
              {countrySearch[0].name.common}
            </p>
            <p>
              <span>Capital</span>
              {countrySearch[0].capital}
            </p>
            <p>
              <span>Sub Region</span>
              {countrySearch[0].subregion}
            </p>
            <p>
              <span>Population</span>
              {countrySearch[0].population}
            </p>
            <p>
              <span>Languages</span>
              {Object.values(countrySearch[0].languages)
                .map((item) => item)
                .join(',')}
            </p>
            <p>
              <span>Currencies</span>
              {Object.values(countrySearch[0].currencies)
                .map((item) => item.name)
                .join(',')}
            </p>
            <p>
              <span>Time Zone</span>
              {countrySearch[0].timezones[0]}
            </p>
            <p>
              <span>Area</span>
              {countrySearch[0].area}
            </p>
            <p>
              <span>Map</span>
              <a
                href={countrySearch[0].maps.googleMaps}
                style={{ color: '#1d505b' }}
              >
                Google Map
              </a>
            </p>
            <p>
              <span>Latitude & Longitude</span>
              {countrySearch[0].latlng.join(' / ')}
            </p>
          </div>
        </div>
      ) : (
        <h1>There No Detail About This Country</h1>
      )}
    </div>
  );
};

export default Details;
