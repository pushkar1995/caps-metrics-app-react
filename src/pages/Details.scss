.detail__container {
    display: flex;
    flex-direction: column;
    color: #ebebeb;

    .detail__header {
        display: flex;
        align-items: center;
        justify-content: space-between;
        background-color: #3ea4b4;
        padding: 1rem;
        gap: 1rem;
        text-align: end;

        img {
            height: 120px;
            width: 200px;
            border-radius: 10px;
            box-shadow: 0 0 10px #0008;
        }

        h2 {
            font-family: cursive;
            font-size: 26px;
        }
    }

    .detail__description {
        background-color: #3ea4b4;

        p {
            padding: 1rem;
            font-size: 18px;
            display: flex;
            justify-content: space-between;
        }

        p:nth-child(odd) {
            background-color: #2c93af;
        }
    }
}