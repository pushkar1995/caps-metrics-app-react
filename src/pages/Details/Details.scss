.detail__container {
  display: flex;
  flex-direction: column;
  color: var(--white-color);

  .detail__header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: var(--secondary-color);
    padding: 1rem;
    gap: 1rem;
    text-align: end;

    img {
      height: 120px;
      width: 200px;
      border-radius: 10px;
      box-shadow: 0 0 10px #0008;
    }

    h2 {
      font-family: cursive;
      font-size: 26px;
    }
  }

  .detail__description {
    background-color: var(--secondary-color);

    p {
      padding: 1rem;
      font-size: 18px;
      display: flex;
      justify-content: space-between;
    }

    p:nth-child(odd) {
      background-color: var(--dark-secondary-color);
    }
  }
}
