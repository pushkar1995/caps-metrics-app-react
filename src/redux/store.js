import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countriesSlice/countriesSlice';

const store = configureStore({
  reducer: {
    country: countriesReducer,
  },
});

export default store;
