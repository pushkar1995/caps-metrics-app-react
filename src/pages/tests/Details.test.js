import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Details from '../Details/Details';
import store from '../../redux/store';

describe('Test the Details', () => {
  it('should match to snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should return the number of p tags', () => {
    render(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    );
    const h = screen.getAllByRole('heading');
    expect(h).toHaveLength(2);
  });
});
