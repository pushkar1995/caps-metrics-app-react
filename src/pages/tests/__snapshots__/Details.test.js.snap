// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Details should match to snapshot 1`] = `
<div>
  <nav
    className="nav__div"
  >
    <a
      className="nav__link"
      href="/"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M17.77 3.77L16 2 6 12l10 10 1.77-1.77L9.54 12z"
        />
      </svg>
    </a>
    <h1
      className="nav__title"
    >
      Details
    </h1>
    <div
      className="nav__icons"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        />
      </svg>
      <svg
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </div>
  </nav>
  <h1>
    There No Detail About This Country
  </h1>
</div>
`;
